### PR TITLE
Fix serialization of broken objects in the driver logs

### PR DESCRIPTION
**Pull request by bigmontz**
Trying to logging broken objects was causing the driver fails in unexpected moments.
This broken behaviour makes the driver fail different when the `debug` log is enabled.

### CODE REVIEW

File: packages/core/src/internal/object-util.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const __isBrokenObject__ = '__isBrokenObject__'
+// eslint-disable-next-line @typescript-eslint/naming-convention
+const __reason__ = '__reason__'
+
+/**
+ * Creates a object which all method call will throw the given error
```
Comment (robsdedude) [inline packages/core/src/internal/object-util.ts]:
```suggestion
 * Creates a object on which all method calls will throw the given error
```

File: packages/core/src/internal/object-util.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  return new Proxy(object, {
+    get: (_: T, p: string | Symbol): any => {
+      if (p === __isBrokenObject__) {
+        return true
+      } else if (p === __reason__) {
+        return error
+      } else if (p === 'toJSON') {
+        return undefined
+      }
```
Comment (robsdedude) [inline packages/core/src/internal/object-util.ts:44]: See [Slack discussion (team internal link)](https://neo4j.slack.com/archives/G493NJPF0/p1662376150650769).

Comment (robsdedude) [reply packages/core/src/internal/object-util.ts:44]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify

File: packages/core/test/json.test.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { json, newError } from '../src'
+import { createBrokenObject } from '../src/internal/object-util'
+
+describe('json', () => {
```
Comment (robsdedude) [inline packages/core/test/json.test.ts:23]: Might be worth opening another PR for adding tests to check the custom json serializer also works with bigint (if the tests don't already exist elsewhere).

**Review by robsdedude — APPROVED** [review]
:feather: :page_facing_up:

File: packages/core/test/json.test.ts (hunk 3 above)
Comment (bigmontz) [reply packages/core/test/json.test.ts:23]: Done

Comment (robsdedude) [reply packages/core/test/json.test.ts:23]: :rocket:
